Sources/Support/PurchasesDiagnostics.swift

```diff
@@ -187,17 +173,68 @@ extension PurchasesDiagnostics {
     /// Status of the SDK Health report
     public enum SDKHealthStatus: Sendable {
         /// SDK configuration is valid but might have some non-blocking issues
-        case healthy(warnings: [PurchasesDiagnostics.Error])
+        case healthy(warnings: [PurchasesDiagnostics.SDKHealthError])
         /// SDK configuration is not valid and has issues that must be resolved
-        case unhealthy(PurchasesDiagnostics.Error)
+        case unhealthy(PurchasesDiagnostics.SDKHealthError)
     }
+}
+#endif
 
-    #if DEBUG
-    /// Performs a full SDK configuration health check and throws an error if the configuration is not valid.
+extension PurchasesDiagnostics {
+
+    /// An error that represents a failing step in ``PurchasesDiagnostics``
+    public enum Error: Swift.Error {
+
+        /// Connection to the API failed
+        case failedConnectingToAPI(Swift.Error)
+
+        /// API key is invalid
+        case invalidAPIKey
+
+        /// Fetching offerings failed due to the underlying error
+        case failedFetchingOfferings(Swift.Error)
+
+        /// Failure performing a signed request
+        case failedMakingSignedRequest(Swift.Error)
+
+        /// Any other not identifier error. You can check the undelying error for details.
+        case unknown(Swift.Error)
+
+    }
+
+}
+
+extension PurchasesDiagnostics {
+
+    /// Checks if the SDK is configured correctly.
     /// - Important: This method is intended solely for debugging configuration issues with the SDK implementation.
     /// It should not be invoked in production builds.
-    /// - Throws: The specific configuration issue that needs to be solved.
+    /// - Throws: ``PurchasesDiagnostics/Error`` if any step fails
+    @available(*, deprecated, message: """
+    Use the `PurchasesDiagnostics.shared.checkSDKHealth()` method instead.
+    """)
+    @objc(testSDKHealthWithCompletion:)
     public func testSDKHealth() async throws {
+        do {
+            try await self.unauthenticatedRequest()
+            #if !ENABLE_CUSTOM_ENTITLEMENT_COMPUTATION
+            try await self.authenticatedRequest()
+            #endif
+            try await self.offeringsRequest()
+            try await self.signatureVerification()
+        } catch let error as Error {
+            throw error
+        } catch let error {
+            // Catch every other error to ensure that we only throw `Error`s from here.
+            throw Error.unknown(error)
+        }
+    }
+
+    #if DEBUG
+    /// Performs a full SDK configuration health check and throws an error if the configuration is not valid.
+    /// - Important: This method can not be invoked in production builds.
+    /// - Throws: The specific configuration issue that needs to be solved.
```

Comment:
Nit here. We should perhaps indicate the type of the error thrown
```suggestion
    /// - Throws: ``SDKHealthError`` indicating the specific configuration issue that needs to be solved.
```